url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/view/ResultsController.java

```diff
@@ -0,0 +1,396 @@
+
+				//String text = names.get(i);
+				String text = parseNames(names.get(i));
+
+
+				String url = "http://google.com/search?btnI=1&q=site%3Ahttp%3A//www.ratemyprofessors.com/ShowRatings.jsp%3Ftid%3D%20%22York%20College%20of%20Pennsylvania%22" + text;
+
+				text = text.replace("%20", " ").trim();
+
+
+				try {
+
+					SimpleAttributeSet attr = new SimpleAttributeSet();
+					attr.addAttribute(StyleConstants.CharacterConstants.Underline, Boolean.TRUE);
+					attr.addAttribute(StyleConstants.CharacterConstants.Foreground, Color.BLUE);
+
+
+					SimpleAttributeSet hrefAttr = new SimpleAttributeSet();
+					hrefAttr.addAttribute(HTML.Attribute.HREF, url.toString());
+
+
+
+					SimpleAttributeSet attrs = new SimpleAttributeSet();
+					attrs.addAttribute(HTML.Tag.A, hrefAttr);
+
+					// Add the text along with its attributes.
+
+					int offset = doc.getLength();
+					int length = text.length();
+
+					doc.insertString(doc.getLength(), text, attr);
+
+
+					doc.setCharacterAttributes(offset, length, attrs, false);
+					//doc.setCharacterAttributes(offset, length, attr, false);
+
+
+>>>>>>> .r125
+				}
+<<<<<<< .mine
+
+=======
+				catch (BadLocationException e) {
+					e.printStackTrace(System.err);
+				}        
+
+			}
+
+			try {
+				doc.insertString(doc.getLength(), "  ", regular);
+>>>>>>> .r125
+			} catch (BadLocationException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+
+	}
+	private String parseNames(String val) {
+		// TODO Auto-generated method stub
+
+		String name = "%20";
+		//name = name.replace("(P)", "");
+
+
+
+		StringTokenizer s = new StringTokenizer(val.replace("(P)", "").trim());
+		if (s.countTokens() == 3)
+		{
+			name += s.nextToken();
+			s.nextToken(); // the middle initial skip...
+			name += "%20" + s.nextToken();
+		}
+
+		return name;
+	}
+<<<<<<< .mine
+	private ArrayList<String> parseNames(String val) {
+		// TODO Auto-generated method stub
+
+		ArrayList<String> names = new ArrayList<String>();
+		names.add(new String());
+
+		StringTokenizer s = new StringTokenizer(val);
+		while(s.hasMoreTokens())
+		{
+			String curr = s.nextToken();
+			if (curr.contains("(") && s.hasMoreTokens());
+			{
+				String next = s.nextToken();
+				if (next.contains(",") && s.hasMoreTokens())
+				{
+					names.add(new String());
+				}
+			}
+
+			if (curr.contains("."))
+			{
+
+			}
+			else
+			{
+				names.set((names.size() - 1), names.get((names.size() - 1)) + "%20" + curr);
+			}
+		}
+
+
+		return names;
+	}
+
+=======
+
+*/
+
+	public void print(JTextPane printTxtPane) {
+		// TODO Auto-generated method stub
+		try {
+			printTxtPane.print();
+
+		} catch (PrinterException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, "Unable to print.", "Error",JOptionPane.ERROR_MESSAGE);
+		}
+
+	}
+
+	public void save(File f, String text)
+	{
+		try {
+			java.io.FileWriter wr = new java.io.FileWriter(f);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *wr*. The resource is referenced at lines 347, 348. The resource is closed at line 349. There are other execution paths that do not contain closure statements, for example, when OutputStreamWriter.flush throws an exception. Ensure *wr* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).